models/articles.models.js

```diff
@@ -1,22 +1,51 @@
 const db = require("../db/connection.js");
 
-exports.fetchArticles = () => {
-  const queryStr = `
+exports.fetchArticles = (sort_by = "created_at", order = "desc", topic) => {
+  const validSortBy = [
+    "article_id",
+    "title",
+    "topic",
+    "author",
+    "body",
+    "created_at",
+    "votes",
+  ];
+  const validOrder = ["asc", "ASC", "desc", "DESC"];
+  // const validTopics = 
+  const topicVal = [];
+  
+  let queryStr = `
   SELECT 
     articles.article_id,
     articles.title,
     articles.topic,
     articles.author,
     articles.created_at,
     articles.votes,
-    COUNT(comment_id)::int AS comment_count
-  FROM
-    articles
+  COUNT(comment_id)::int AS comment_count
+  FROM articles
   LEFT JOIN comments
-  ON articles.article_id = comments.article_id
-  GROUP BY articles.article_id
-  ORDER BY created_at DESC`;
-  return db.query(queryStr).then(({ rows }) => {
+  ON articles.article_id = comments.article_id`;
+
+  if (topic) {
+    queryStr += ` WHERE topic = $1 
+                  GROUP BY articles.article_id`;
+    topicVal.push(topic);
+  } else {
+    queryStr += ` GROUP BY articles.article_id`;
+  }
+
```

Comment:
To DRY up this code ... you could add the GROUP BY section of the query once, outside of the if else, as you are always wanting to add this section, regardless of whether a topic has been queried or not